### PR TITLE
linux-sgx: implement POSIX calls based on getsockname and set/getbooloption.

### PR DESCRIPTION
**Pull request by JamesMenetrey**
Hi there,

I continue implementing some of the missing POSIX OCALLs that I use in my apps.

The ones I propose in this PR are the OCALLs that rely on the POSIX function `getbooloption`, `setbooloption` and `getsockname `.

Cheers

### CODE REVIEW

File: core/shared/platform/linux-sgx/sgx_socket.c [hunk 1]
```diff
@@ -905,49 +966,53 @@ os_socket_get_recv_buf_size(bh_socket_t socket, size_t *bufsiz)
...
+    return os_socket_getbooloption(socket, SOL_SOCKET, SO_REUSEADDR,
+                                   is_enabled);
 }
 
 int
 os_socket_set_reuse_port(bh_socket_t socket, bool is_enabled)
 {
-    errno = ENOSYS;
-
+#if defined(SO_REUSEPORT) /* NuttX doesn't have SO_REUSEPORT */
```
Comment (wenyongh) [inline core/shared/platform/linux-sgx/sgx_socket.c:997]: SO_REUSEPORT is defined in sgx_socket.h，shall we remove the macro control here?

File: core/shared/platform/linux-sgx/sgx_socket.c [hunk 2]
```diff
@@ -905,49 +966,53 @@ os_socket_get_recv_buf_size(bh_socket_t socket, size_t *bufsiz)
...
     return BHT_ERROR;
+#endif /* defined(SO_REUSEPORT) */
 }
 
 int
 os_socket_get_reuse_port(bh_socket_t socket, bool *is_enabled)
 {
-    errno = ENOSYS;
-
+#if defined(SO_REUSEPORT) /* NuttX doesn't have SO_REUSEPORT */
```
Comment (wenyongh) [inline core/shared/platform/linux-sgx/sgx_socket.c:1009]: ditto

File: core/shared/platform/linux-sgx/sgx_socket.c [hunk 3]
```diff
@@ -969,33 +1034,41 @@ os_socket_get_linger(bh_socket_t socket, bool *is_enabled, int *linger_s)
...
-
-    return BHT_ERROR;
+    return os_socket_getbooloption(socket, IPPROTO_TCP, TCP_NODELAY,
+                                   is_enabled);
 }
 
 int
 os_socket_set_tcp_quick_ack(bh_socket_t socket, bool is_enabled)
 {
+#ifdef TCP_QUICKACK
```
Comment (wenyongh) [inline core/shared/platform/linux-sgx/sgx_socket.c:1051]: ditto

**Comment by JamesMenetrey** [comment]
Hello @wenyongh,

Thanks for your review! I have applied the changes. I also noticed it was the case for `TCP_FASTOPEN_CONNECT`, so I have removed the `#if`/`#endif` clauses related to it.

Cheers

**Pull request merged**